packages/react-native-session-replay/android/src/main/kotlin/com/datadog/reactnative/sessionreplay/utils/ReflectionUtils.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.reactnative.sessionreplay.utils
+
+import com.datadog.android.api.InternalLogger
+import java.lang.reflect.Field
+
+internal class ReflectionUtils(
+    private val logger: InternalLogger
+) {
+    internal fun getDeclaredField(className: Class<*>, instance: Any?, fieldName: String): Any? {
+        val declaredField = searchForField(className, fieldName) ?: return null
+
+        declaredField.let {
+            it.isAccessible = true
+            return it.get(instance)
+        }
+    }
+
+    private fun searchForField(className: Class<*>, fieldName: String): Field? {
+        try {
+            return className.getDeclaredField(fieldName)
+        } catch (e: NoSuchFieldException) {
+            if (className.superclass != null) {
+                searchForField(className.superclass, fieldName)
+            } else {
```

Comment:
we shouldn't rely on the exception to do the lookup in the `super` class. Instead, can do `getDeclaredField().firstOrNull { it.name == fieldName }` and if result is `null` and there is no `super` class, then return `null`, otherwise if not found, check `super` class.

Exceptions should stays exceptions.